core_utilities/archive_checker.py

```diff
@@ -0,0 +1,451 @@
+"""Application for checking the status of videos in the pony archive."""
+
+import tkinter as tk
+import csv
+import threading
+import requests
+import math
+import io
+import asyncio
+import aiohttp
+import json
+import os
+
+from tkinter import filedialog, Event, ttk
+from PIL import ImageTk, Image
+from typing import List, Tuple
+from yt_dlp import YoutubeDL, DownloadError
+from enum import Enum
+from classes.gui import GUI
+
+class ArchiveIndices:
+    LINK = 3
+    TITLE = 4
+    CHANNEL = 5
+    STATE = 7
+    ALT_LINK = 8
+    FOUND = 9
+    NOTES = 10
+
+class States(Enum):
+    NON_EMBEDDABLE = ('non-embedable', 'non-embeddable')
+    UNAVAILABLE = ('unavailable', 'deleted', 'private', 'tos deleted')
+    AGE_RESTRICTED = ('age-restricted',)
+    BLOCKED = ('blocked',)
+
+    @classmethod
+    def get(cls, value: str):
+        for state in cls:
+            if value.lower() in state.value:
+                return state
+
+blocked_everywhere_indicator = 'EVERYWHERE EXCEPT:'
+video_not_found = "Video not found"
+lock = asyncio.Lock()
```

Comment:
Maybe move these three globals above the classes, so that they're easier for devs to spot